esp/smc/SMCLib/TpContainer.cpp

```diff
@@ -756,12 +781,22 @@ extern TPWRAPPER_API void validateTargetName(const char* target)
     if (isEmptyString(target))
         throw makeStringException(ECLWATCH_INVALID_CLUSTER_NAME, "Empty target name.");
 
-    CriticalBlock block(validTargetSect);
+    CriticalBlock block(configUpdateSect);
     configUpdateHook.installOnce(configUpdate, true);
     if (validTargets.find(target) == validTargets.end())
         throw makeStringExceptionV(ECLWATCH_INVALID_CLUSTER_NAME, "Invalid target name: %s", target);
 }
 
+extern TPWRAPPER_API bool validateDataPlaneName(const char * remoteDali, const char * name)
+{
+    if (isEmptyString(name))
+        throw makeStringException(ECLWATCH_INVALID_CLUSTER_NAME, "Empty data plane name.");
+
+    CriticalBlock block(configUpdateSect);
+    configUpdateHook.installOnce(configUpdate, true);
```

Comment:
better if the installOnce is outside the critical block - same for the call above on line 785

Reply:
Ok, I see that installOnce() is thread safe.